session.go

```diff
@@ -388,18 +394,29 @@ func (s *session) Execute(sql string) ([]rset.Recordset, error) {
 	var rs []rset.Recordset
 
 	for _, st := range statements {
+		inHistory := false
+		if s.ShouldAutocommit(s) {
+			if isPreparedStmt(st) {
+				ps := st.(*stmts.PreparedStmt)
+				s.history.add(ps.ID, st)
+			} else {
+				s.history.add(0, st)
+			}
+			inHistory = true
```

Comment:
We should retry some stmt after we get any retryable error. Those errors are thrown by session.FinishTxn(false). 
If the ctx is autocommit, session.FinishTxn() will called automaticly. So we should add stmt into history before executing it. In otherwise, the history list will miss this stmt. Please check the implementation of session.FinishTxn().